src/System.IO.FileSystem.AccessControl/src/System/Security/AccessControl/FileSecurity.cs

```diff
@@ -0,0 +1,666 @@
+                fileSystemRights,
+                InheritanceFlags.None,
+                PropagationFlags.None,
+                flags)
+        {
+        }
+
+        public FileSystemAuditRule(
+            IdentityReference identity,
+            FileSystemRights fileSystemRights,
+            InheritanceFlags inheritanceFlags,
+            PropagationFlags propagationFlags,
+            AuditFlags flags)
+            : this(
+                identity,
+                AccessMaskFromRights(fileSystemRights),
+                false,
+                inheritanceFlags,
+                propagationFlags,
+                flags)
+        {
+        }
+
+        public FileSystemAuditRule(
+            String identity,
+            FileSystemRights fileSystemRights,
+            AuditFlags flags)
+            : this(
+                new NTAccount(identity),
+                fileSystemRights,
+                InheritanceFlags.None,
+                PropagationFlags.None,
+                flags)
+        {
+        }
+
+        public FileSystemAuditRule(
+            String identity,
+            FileSystemRights fileSystemRights,
+            InheritanceFlags inheritanceFlags,
+            PropagationFlags propagationFlags,
+            AuditFlags flags)
+            : this(
+                new NTAccount(identity),
+                AccessMaskFromRights(fileSystemRights),
+                false,
+                inheritanceFlags,
+                propagationFlags,
+                flags)
+        {
+        }
+
+        internal FileSystemAuditRule(
+            IdentityReference identity,
+            int accessMask,
+            bool isInherited,
+            InheritanceFlags inheritanceFlags,
+            PropagationFlags propagationFlags,
+            AuditFlags flags)
+            : base(
+                identity,
+                accessMask,
+                isInherited,
+                inheritanceFlags,
+                propagationFlags,
+                flags)
+        {
+        }
+
+        #endregion
+
+        #region Private methods
+
+        private static int AccessMaskFromRights(FileSystemRights fileSystemRights)
+        {
+            if (fileSystemRights < (FileSystemRights)0 || fileSystemRights > FileSystemRights.FullControl)
+                throw new ArgumentOutOfRangeException("fileSystemRights", SR.Format(SR.Argument_InvalidEnumValue, fileSystemRights, "FileSystemRights"));
+            Contract.EndContractBlock();
+
+            return (int)fileSystemRights;
+        }
+
+        #endregion
+
+        #region Public properties
+
+        public FileSystemRights FileSystemRights
+        {
+            get { return FileSystemAccessRule.RightsFromAccessMask(base.AccessMask); }
+        }
+        #endregion
+    }
+
+
+    public abstract class FileSystemSecurity : NativeObjectSecurity
+    {
+        #region Member variables
+
+        private const ResourceType s_ResourceType = ResourceType.FileObject;
```

Comment:
Nit: constants should be named PascalCase, e.g. ```private const ResourceType FileObjectResourceType```.  That said, in a case like this, why is this even needed at all... call sites can just use ```ResoureType.FileObject``` directly.